File: Challenge/ANAGRAMA/anagrama.kt
fun sonAnagramas(palabra1: String, palabra2: String): Boolean {
    // Convertimos las palabras a minúsculas y eliminamos espacios en blanco
    val palabra1 = palabra1.toLowerCase().replace(" ", "")
    val palabra2 = palabra2.toLowerCase().replace(" ", "")

    // Verificamos si las palabras tienen la misma longitud
    if (palabra1.length != palabra2.length) {
        return false
    }

    // Ordenamos las letras de ambas palabras y comparamos
    val sortedPalabra1 = palabra1.toCharArray().sorted().joinToString("")
    val sortedPalabra2 = palabra2.toCharArray().sorted().joinToString("")

    return sortedPalabra1 == sortedPalabra2
}

fun main() {
    // Ejemplos de uso
    val resultado1 = sonAnagramas("listen", "silent")
    println(resultado1)  // true

    val resultado2 = sonAnagramas("hello", "world")
    println(resultado2)  // false
}

File: Challenge/ANAGRAMA/anagrama.swift
func sonAnagramas(_ palabra1: String, _ palabra2: String) -> Bool {
    // Convertimos las palabras a minúsculas y eliminamos espacios en blanco
    let palabra1 = palabra1.lowercased().replacingOccurrences(of: " ", with: "")
    let palabra2 = palabra2.lowercased().replacingOccurrences(of: " ", with: "")

    // Verificamos si las palabras tienen la misma longitud
    guard palabra1.count == palabra2.count else {
        return false
    }

    // Ordenamos las letras de ambas palabras y comparamos
    let sortedPalabra1 = String(palabra1.sorted())
    let sortedPalabra2 = String(palabra2.sorted())

    return sortedPalabra1 == sortedPalabra2
}

// Ejemplos de uso
let resultado1 = sonAnagramas("listen", "silent")
print(resultado1)  // true

let resultado2 = sonAnagramas("hello", "world")
print(resultado2)  // false
